riscv/cfg.h

```diff
@@ -25,24 +26,46 @@ class cfg_arg_t {
   bool was_set;
 };
 
+// Configuration that describes a memory region
+class mem_cfg_t
+{
+public:
+  mem_cfg_t(reg_t base, reg_t size)
+    : base(base), size(size)
+  {
+    // The truth of these assertions should be ensured by whatever is creating
+    // the regions in the first place, but we have them here to make sure that
+    // we can't end up describing memory regions that don't make sense. They
+    // ask that size is a multiple of 32 bits, that it is positive and that the
+    // top address is still representable in a reg_t.
+    assert((size % 4 == 0) && (base + size > base));
```

Comment:
We actually need to enforce 4 KiB size and 4 KiB alignment because of the implementation of `mmu_t`.  I agree with your comment that other code should be responsible for that sort of input validation, but if we're going to have an assertion here, we should assert the same thing.

Reply:
You're completely right: I have no idea where that 4 came from! (Maybe I mis-read the code in `mem_t::mem_t`?) Fixing now.